.github/workflows/nms-workflow.yml

```diff
@@ -146,7 +146,7 @@ jobs:
       - name: run yarn test
         run: |
           cd ${MAGMA_ROOT}/nms
-          yarn add jest@^26.4.2 -W --dev
+          yarn add jest@^28.1.3 -W --dev
```

Comment:
There's a version mismatch of jest in the CI pipeline, I'm not sure if this is causing the test errors but `jest.useFakeTimers();` is valid code and should be supported as it's required for unit testing the rate limiter. 

Jest 26 should support this as well and the error I'm seeing is: https://github.com/magma/magma/actions/runs/8312668614/job/22990589385
```
 Summary of all failing tests
FAIL server/middleware/__tests__/rateLimit-test.ts
  ● Test suite failed to run

    TypeError: Cannot assign to read only property 'performance' of object '[object global]'
```
I'll attempt to use Jest 29 in the CI if this still happens as it seems to have helped in this issue https://github.com/jestjs/jest/issues/14741

Reply:
We discussed the jest upgrade in the last TSC and had consensus to go with it.

Reply:
It does not seem to have worked, I'm not sure I fully get why the CI prevents us from replacing the timer implementation at the moment. If anyone has an idea as to why that might be the case here feel free to chime in, here's the error:
```
Summary of all failing tests
FAIL server/middleware/__tests__/rateLimit-test.ts
  ● Test suite failed to run

    TypeError: Cannot assign to read only property 'performance' of object '[object global]'

      27 | };
      28 |
    > 29 | jest.useFakeTimers();
         |      ^
      30 | jest.mock('../../../config/config', () => config);
      31 |
      32 | describe('Rate limit test', () => {

      at hijackMethod (node_modules/@sinonjs/fake-timers/src/fake-timers-src.js:946:32)
      at Object.install (node_modules/@sinonjs/fake-timers/src/fake-timers-src.js:1733:17)
      at FakeTimers.useFakeTimers (node_modules/@jest/fake-timers/build/modernFakeTimers.js:110:36)
      at Object.useFakeTimers (server/middleware/__tests__/rateLimit-test.ts:29:6)


Test Suites: 1 failed, 4 skipped, 53 passed, 54 of 58 total
Tests:       4 skipped, 351 passed, 355 total
Snapshots:   0 total
Time:        133.391 s
Ran all test suites in 2 projects.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.
```